components/tabs/src/storage.rs

```diff
@@ -399,6 +399,13 @@ fn compute_serialized_size(v: &Vec<RemoteTab>) -> usize {
     serde_json::to_string(v).unwrap_or_default().len()
 }
 
+fn safe_truncate(s: &str, max_chars: usize) -> &str {
+    match s.char_indices().nth(max_chars) {
+        None => s,
+        Some((idx, _)) => &s[..idx],
+    }
```

Comment:
I think this truncates things based on unicode chars, but you want to be truncating based on the `idx` values.

Reply:
hmm I'm not sure if this is quite right, Rust treats all strings as UTF8 https://doc.rust-lang.org/book/ch08-02-strings.html#what-is-a-string but i'll have to double check on this